core/trino-main/src/main/java/io/trino/operator/join/LookupJoinPageBuilder.java

```diff
@@ -98,44 +100,50 @@ public void appendNullForBuild(JoinProbe probe)
         }
     }
 
+    private static Block unwrapLoadedBlocks(Block filteredProbeBlock)
+    {
+        // Lazy blocks (e.g used in filter condition) could be loaded during filter evaluation.
+        // Unwrap them to reduce overhead of further processing.
+        return filteredProbeBlock.isLoaded() ? filteredProbeBlock.getLoadedBlock() : filteredProbeBlock;
```

Comment:
`filteredProbeBlock.getLoadedBlock()` should be noop (return same instance) when block is already loaded, hence check to `filteredProbeBlock.isLoaded()` is not needed

Reply:
The idea with the previous logic here (that I attempted to preserve) is that lazy blocks that _aren't_ loaded shouldn't be force loaded- only lazy blocks that _are_ loaded but still wrap their underlying (now loaded) block.